server/shared/publishing/helpers.js

```diff
@@ -151,23 +151,39 @@ function publishAssessments(parent) {
 
 function fetchContainers(parent) {
   const typeConfigs = getSupportedContainers(parent.type);
-  const isCore = it => !containerRegistry.getStaticsResolver(it.type);
-  const coreTypes = typeConfigs.filter(isCore).map(it => it.type);
+  const { defaults, custom } = getGroupedContainerTypes(typeConfigs);
 
-  return Promise.all([
-    parent.getChildren({ where: { type: coreTypes } }).map(fetchDefaultContainer),
-    fetchCustomContainers(parent)
-  ])
-  .reduce((containers, groupedContainers) => {
+  const fetches = [];
+  if (defaults.length) fetches.push(fetchDefaultContainers(parent, defaults));
+  if (custom.length) fetches.push(fetchCustomContainers(parent, custom));
+  if (!fetches.length) return [];
+  return Promise.all(fetches).reduce((containers, groupedContainers) => {
     const mappedContainers = groupedContainers.map(it => {
       const config = find(typeConfigs, { type: it.type });
       const publishedAs = get(config, 'publishedAs', 'container');
-      return { ...it, publishedAs };
+      const { templateId } = config;
+      return { ...it, templateId, publishedAs };
     });
     return containers.concat(mappedContainers);
   }, []);
 }
 
+function getGroupedContainerTypes(typeConfigs) {
+  return typeConfigs.reduce((acc, { type, templateId }) => {
+    if (containerRegistry.getContentFetcher(templateId)) {
+      acc.custom.push(templateId);
+    } else {
+      acc.defaults.push(type);
+    }
+    return acc;
+  }, { defaults: [], custom: [] });
+}
+
+function fetchDefaultContainers(parent, types) {
+  const where = { type: types };
+  return parent.getChildren({ where }).map(fetchDefaultContainer);
```

Comment:
It feels like we can turn this into one query 🤔

Reply:
True. If we pass a Content Elements include to the `getChildren()` method we can fetch them all at once.